tests/scripts/thread-cert/test_srp_lease.py

```diff
@@ -0,0 +1,191 @@
+
+# Test description:
+#   This test verifies the SRP client & server properly handle SRP host
+#   and service instance lease.
+#
+# Topology:
+#     LEADER (SRP server)
+#       |
+#       |
+#     ROUTER (SRP client)
+#
+
+SERVER = 1
+CLIENT = 2
+LEASE = 10  # Seconds
+KEY_LEASE = 20  # Seconds
+
+
+class SrpRegisterSingleService(thread_cert.TestCase):
+    USE_MESSAGE_FACTORY = False
+    SUPPORT_NCP = False
+
+    TOPOLOGY = {
+        SERVER: {
+            'name': 'SRP_SERVER',
+            'masterkey': '00112233445566778899aabbccddeeff',
+            'mode': 'rdn',
+            'panid': 0xface
+        },
+        CLIENT: {
+            'name': 'SRP_CLIENT',
+            'masterkey': '00112233445566778899aabbccddeeff',
+            'mode': 'rdn',
+            'panid': 0xface,
+            'router_selection_jitter': 1
+        },
+    }
+
+    def test(self):
+        server = self.nodes[SERVER]
+        client = self.nodes[CLIENT]
+
+        #
+        # 0. Start the server & client devices.
+        #
+
+        server.srp_server_set_enabled(True)
+        server.srp_server_set_lease_range(LEASE, LEASE, KEY_LEASE, KEY_LEASE)
+        server.start()
+        self.simulator.go(5)
+        self.assertEqual(server.get_state(), 'leader')
+        self.simulator.go(5)
+
+        client.srp_server_set_enabled(False)
+        client.start()
+        self.simulator.go(5)
+        self.assertEqual(client.get_state(), 'router')
+
+        #
+        # 1. Register a single service and verify that it works.
+        #
+
+        client.srp_client_set_host_name('my-host')
+        client.srp_client_set_host_address('2001::1')
+        client.srp_client_start(server.get_addrs()[0], client.get_srp_server_port())
+        client.srp_client_add_service('my-service', '_ipps._tcp', 12345)
+        self.simulator.go(2)
+
+        self.check_host_and_service(server, client)
+
+        #
+        # 2. Stops the client and waits for the service instance LEASE to expire.
+        #
+
+        client.srp_client_stop()
+        self.simulator.go(LEASE + 1)
+
+        # The SRP server should remove the host & service but retain their names
+        # since the the KEY LEASE doesn't expire yet.
+        self.assertEqual(server.srp_server_get_host('my-host')['deleted'], 'true')
+        self.assertEqual(server.srp_server_get_service('my-service', '_ipps._tcp')['deleted'], 'true')
+
+        # Start the client again, the same service should be successfully registered.
+        client.srp_client_start(server.get_addrs()[0], client.get_srp_server_port())
+        self.simulator.go(2)
+
+        self.check_host_and_service(server, client)
+
+        #
+        # 3. Stop the client and wait for the KEY LEASE to expire. The SRP server
+        #    should permanently remove the host & service instance.
+        #
+
+        client.srp_client_stop()
+        self.simulator.go(KEY_LEASE + 1)
+
+        # The host & service are expected to be permanently removed.
```

Comment:
```suggestion
        # The host & service are expected to be removed.
```

Reply:
There are two removed/deleted status, one is the service/host is removed but the name is retained. And the other is the service/host and its name are both removed. I was calling the second `permanently removed` and changed it to `fully removed` at here.